lib/model/CResourceMonitor.cc

```diff
@@ -311,9 +313,14 @@ void CResourceMonitor::decreaseMargin(core_t::TTime elapsedTime) {
     // will be the overwhelmingly common source of additional memory
     // so the model memory should be accurate (on average) in this
     // time frame.
-    double scale{1.0 - static_cast<double>(elapsedTime) /
-                           static_cast<double>(core::constants::DAY)};
-    m_ByteLimitMargin = 1.0 - scale * (1.0 - m_ByteLimitMargin);
+    if (elapsedTime <= core::constants::DAY) {
+        double scale{1.0 - static_cast<double>(elapsedTime) /
+                               static_cast<double>(core::constants::DAY)};
```

Comment:
Good catch!

I think a cleaner fix is just to lower bound `elapsedTime`, i.e. something like `static_cast<double>(std::min(elapsedTime, x)) / static_cast<double>(core::constants::DAY)`. This way the rate at which the margin is decreased is a smooth function of the bucket length. As it is now there is a very big discontinuity when the bucket length is changed from DAY to DAY + eps.

The I'd recommended choosing a value for `x` around 2 hours. For longer bucket lengths the common source of extra memory will be weekly periodicity. This takes longer to detect. We want the`m_ByteLimitMargin` to approach 1 on this time scale for long bucket lengths and this limit would achieves this.

Also, I think the comment could be clarified a bit. We'll aim for `m_ByteLimitMargin` * target memory. So whilst we do want to increase `m_ByteLimitMargin` by doing this we're effectively decreasing the safety margin we're applying to the process memory. The function name refers to this safety margin whilst the comment refers to `m_ByteLimitMargin`.